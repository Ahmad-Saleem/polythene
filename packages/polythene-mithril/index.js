export * from "polythene-mithril-base";
export * from "polythene-mithril-button";
export * from "polythene-mithril-fab";
export * from "polythene-mithril-icon";
export * from "polythene-mithril-raised-button";
export * from "polythene-mithril-ripple";
export * from "polythene-mithril-shadow";
export * from "polythene-mithril-svg";
