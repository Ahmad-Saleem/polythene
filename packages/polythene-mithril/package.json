{
  "name": "polythene-mithril",
  "version": "1.0.0-rc.1",
  "description": "",
  "main": "dist/polythene-mithril.js",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "npm run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-mithril-base": "1.0.0-rc.1",
    "polythene-mithril-button": "1.0.0-rc.1",
    "polythene-mithril-card": "1.0.0-rc.1",
    "polythene-mithril-checkbox": "1.0.0-rc.1",
    "polythene-mithril-dialog": "1.0.0-rc.1",
    "polythene-mithril-dialog-pane": "1.0.0-rc.1",
    "polythene-mithril-fab": "1.0.0-rc.1",
    "polythene-mithril-icon": "1.0.0-rc.1",
    "polythene-mithril-icon-button": "1.0.0-rc.1",
    "polythene-mithril-list": "1.0.0-rc.1",
    "polythene-mithril-list-tile": "1.0.0-rc.1",
    "polythene-mithril-menu": "1.0.0-rc.1",
    "polythene-mithril-radio-button": "1.0.0-rc.1",
    "polythene-mithril-raised-button": "1.0.0-rc.1",
    "polythene-mithril-ripple": "1.0.0-rc.1",
    "polythene-mithril-shadow": "1.0.0-rc.1",
    "polythene-mithril-svg": "1.0.0-rc.1",
    "polythene-mithril-switch": "1.0.0-rc.1",
    "polythene-mithril-tabs": "1.0.0-rc.1",
    "polythene-mithril-textfield": "1.0.0-rc.1",
    "polythene-mithril-toolbar": "1.0.0-rc.1"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}

