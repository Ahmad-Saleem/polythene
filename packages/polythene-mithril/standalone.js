export * from "./index";
import { addLayoutStyles, addFastClick } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-material-design";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
