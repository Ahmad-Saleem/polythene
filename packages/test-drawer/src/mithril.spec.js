import { runSnapshots } from "../../../scripts/tests/mithril-snapshots";
import { renderer, keys, Drawer } from "polythene-mithril";
import specTests from "./spec-tests.js";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: specTests({ Drawer, renderer, keys }).concat(mithrilTests),
  renderer
});
