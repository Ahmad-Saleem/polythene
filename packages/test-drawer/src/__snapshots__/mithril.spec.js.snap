exports[`test No options 1`] = `
"<span></span>
"
`;

exports[`test Option: class 1`] = `
"<span></span>
"
`;

exports[`test Option: className 1`] = `
"<span></span>
"
`;

exports[`test Option: id 1`] = `
"<span></span>
"
`;

exports[`test Permanent, floating (no shadow) 1`] = `
"<div class=\"pe-dialog pe-drawer pe-drawer--permanent\">
  <div class=\"pe-dialog__touch\"></div>
  <div class=\"pe-dialog__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-0\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-0\"></div>
    </div>
    <form class=
    \"pe-dialog-pane pe-dialog-pane--body-full-bleed pe-drawer pe-drawer--permanent\">
    <div class=\"pe-dialog-pane__content\">
        <div class=\"pe-dialog-pane__body\">
          <div class=\"pe-list pe-list--compact pe-list--padding\">
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M19 3H4.99c-1.11 0-1.98.89-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.11-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Inbox
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Starred
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M2.01 21L23 12 2.01 3 2 10l15 2-15 2z\"></path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Sent mail
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M21.99 8c0-.72-.37-1.35-.94-1.7L12 1 2.95 6.3C2.38 6.65 2 7.28 2 8v10c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2l-.01-10zM12 13L3.74 7.84 12 3l8.26 4.84L12 13z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Drafts
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
"
`;

exports[`test Permanent, floating (shadow depth 1) 1`] = `
"<div class=\"pe-dialog pe-drawer pe-drawer--permanent\">
  <div class=\"pe-dialog__touch\"></div>
  <div class=\"pe-dialog__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <form class=
    \"pe-dialog-pane pe-dialog-pane--body-full-bleed pe-drawer pe-drawer--permanent\">
    <div class=\"pe-dialog-pane__content\">
        <div class=\"pe-dialog-pane__body\">
          <div class=\"pe-list pe-list--compact pe-list--padding\">
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M19 3H4.99c-1.11 0-1.98.89-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.11-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Inbox
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Starred
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M2.01 21L23 12 2.01 3 2 10l15 2-15 2z\"></path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Sent mail
                  </div>
                </div>
              </div>
            </div>
            <div class=
            \"pe-list-tile pe-list-tile--hoverable pe-list-tile--navigation pe-list-tile--front tests-drawer-navigation-list\"
            tabindex=\"0\">
              <div class=\"pe-list-tile__primary\" style=\"\">
                <div class=
                \"pe-list-tile__content pe-list-tile__content-front\">
                <div class=\"pe-icon pe-icon--regular\">
                    <div class=\"pe-svg\">
                      <svg xmlns=\"http://www.w3.org/2000/svg\"
                      width=\"24\" height=\"24\" viewbox=\"0 0 24 24\"
                      focusable=\"false\">
                        <path d=
                        \"M21.99 8c0-.72-.37-1.35-.94-1.7L12 1 2.95 6.3C2.38 6.65 2 7.28 2 8v10c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2l-.01-10zM12 13L3.74 7.84 12 3l8.26 4.84L12 13z\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
                <div class=\"pe-list-tile__content\">
                  <div class=\"pe-list-tile__title\">
                    Drafts
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
"
`;
